### PR TITLE
*: allow setting @@tidb_tso_client_batch_max_wait_time to a float millisecond

### LINKED ISSUE #29698: Allow setting @@tidb_tso_client_batch_max_wait_time to a float millisecond
## Feature Request

**Is your feature request related to a problem? Please describe:**

Allow setting `@@tidb_tso_client_batch_max_wait_time` to a float millisecond. Such as:

```mysql
set global tidb_tso_client_batch_max_wait_time = 0.5
```

### LINKED ISSUE #29722: Value out of the range will only set the session variable leaving PD client unset.
### 1. Minimal reproduce step (Required)

image

### 2. What did you expect to see? (Required)

The PD client sets this truncated value.

### 3. What did you see instead (Required)

The PD client set the original value and failed.

### 4. What is your TiDB version? (Required)

31a4f6abbb2d32460aa844a9641af7c30a7d2cda

### PR DESCRIPTION
**Pull request by JmPotato**
Signed-off-by: JmPotato <ghzpotato@gmail.com>

### What problem does this PR solve?

Issue Number: close #29698, close #29722.

### What is changed and how it works?

* Allow setting @@tidb_tso_client_batch_max_wait_time to a float millisecond.
* Fix the bug that value out of the range will only set the session variable leaving PD client unset.

### Check List

Tests 

- [x] Unit test
- [x] Integration test
- [x] Manual test (add detailed scripts or steps below)

image

### Release note

```release-note
None
```

### CODE REVIEW

File: executor/set.go [hunk 1]
```diff
@@ -204,16 +204,30 @@ func (e *SetExecutor) setSysVariable(ctx context.Context, name string, v *expres
...
+func (e *SetExecutor) checkPDClientDynamicOption(name string, sessionVars *variable.SessionVars) error {
+	if name != variable.TiDBTSOClientBatchMaxWaitTime &&
+		name != variable.TiDBEnableTSOFollowerProxy {
+		return nil
+	}
+	var (
+		err    error
+		valStr string
+	)
+	valStr, err = sessionVars.GlobalVarsAccessor.GetGlobalSysVar(name)
```
Comment (nolouch) [inline executor/set.go:216]: Why cannot use the old `valStr`?

Comment (JmPotato) [reply executor/set.go:216]: Because the valve will be truncated if it's out of range, using the old value will fail when set it to the PD client.

**Review by rleungx — APPROVED** [review]

**Review by nolouch — APPROVED** [review]

**Review by HunDunDM — APPROVED** [review]

**Comment by JmPotato** [comment]
@morgo @bb7133 Could you please help approve this PR? Thx a lot!

**Review by morgo — APPROVED** [review]

**Pull request merged**